indra/db/populate_content.py

```diff
@@ -0,0 +1,705 @@
+from __future__ import absolute_import, print_function, unicode_literals
+from builtins import dict, str
+
+import csv
+import time
+import tarfile
+import zlib
+import logging
+import xml.etree.ElementTree as ET
+import re
+import os
+import multiprocessing as mp
+from os import path
+from ftplib import FTP
+from io import BytesIO
+from indra.util import zip_string, unzip_string
+from indra.util import UnicodeXMLTreeBuilder as UTB
+from indra.literature.pmc_client import id_lookup
+from indra.literature import pubmed_client
+from indra.db import get_aws_db, texttypes
+
+logger = logging.getLogger('update_db')
+
+ftp_blocksize = 33554432 # Chunk size recommended by NCBI
+BATCH_SIZE = 10000
+
+    
+
+
+class NihFtpClient(object):
+    '''
+    High level access to the nih ftp repositories.
+    
+    Parameters
+    ----------
+    my_path: str 
+        The path to the subdirectory around which this client operates.
+    ftp_url: str: (default: 'ftp.ncbi.nlm.nih.gov')
```

Comment:
For optional arguments, `Optional[str]`. Include default values at the end of the arg description.